src/aparna/Shop.java

```diff
@@ -0,0 +1,173 @@
+/*Assignment  7  : 30th March 2021 
+
+1. Create a system (Shop) which maintains each item's quantity.
+
+a. Initial stocks of Maggie: 50 Maggie packets 
+b. Initial stocks of Dosa : 43 packets 
+c. Initial stocks of pouches : 39 oil packets 
+d. Initial stocks of panipuri : 43 packets  
+e. Initial stocks of masala�s : 73 packets.
+
+Expectation:
+1) Only one Customer came for shopping, he picked different items with different quantities, print "running out of stock" message if a particular item is sold out otherwise deduct requested quantity from total quantity. Partial selling is allowed.  
+2) Create a method which shows only those item which is "Out of stock".
+3) Create a method which shows only those item which is "Available in stock" & respective quantities.
+*/
+package aparna;
+
+public class Shop {
+	int totalMaggiePacks, totalDosaPacks, totalOilPouches, totalPanipuriPacks, totalMasalaPacks;
+
+	void initialStocks(int maggiePack, int dosaPack, int oilPouch, int panipuriPack, int masalaPack) {
+		totalMaggiePacks = maggiePack;
+		totalDosaPacks = dosaPack;
+		totalOilPouches = oilPouch;
+		totalPanipuriPacks = panipuriPack;
+		totalMasalaPacks = masalaPack;
+	}
+
+	void requestedQuantity(int reqMaggiePack, int reqDosaPack, int reqOilPouch, int reqPanipuriPack, int reqMasalaPack,
+			boolean partialAllowedStatus) {
+
+		if (totalMaggiePacks == 0)
+			System.out.println("Requested Maggy Packets are out of stock");
+		else if (reqMaggiePack <= totalMaggiePacks) {
+			totalMaggiePacks = totalMaggiePacks - reqMaggiePack;
+		} else {
+			System.out.println("Sorry, not sufficient Maggy Packets quantity available , So we can give you only : "
+					+ totalMaggiePacks + " Maggy Packets Only ");
+			System.out.println("Maggy Packets Actual Quantity is: " + totalMaggiePacks + " and requested Quantity is : "
+					+ reqMaggiePack + " So could not serve the Demand");
+			if (partialAllowedStatus) {
+				totalMaggiePacks = 0;
```

Comment:
If request stock is not available mean we are setting direct it zero, 
what if second time your calling same method and let say you request 20 packs, then this looks not logically correct